devel/gettext/Portfile

```diff
@@ -99,24 +64,59 @@ configure.args-append   --without-cvs \
 test.run                yes
 test.target             check
 
-post-destroot {
-    if {[file exists ${destroot}${myprefix}/lib/charset.alias]} {
-        delete ${destroot}${myprefix}/lib/charset.alias
+subport ${name}-runtime {
+    description             GNU internationalization (i18n) and localization (l10n) runtime library
+
+    long_description        ${subport} is the ${description}, used for writing multilingual programs. \
+                            This is part of GNU gettext. For building applications with gettext \
+                            you need gettext tools provided by the gettext port. \
+                            ${subport} provides the libintl library.
+
+    worksrcdir              ${distname}/gettext-runtime
+
+    post-destroot {
+        delete ${destroot}${prefix}/lib/charset.alias
+        delete ${destroot}${prefix}/share/locale/locale.alias
     }
-    if {[file exists ${destroot}${myprefix}/share/locale/locale.alias]} {
-        delete ${destroot}${myprefix}/share/locale/locale.alias
+}
+
+subport ${name}-libtextstyle {
+    description             ${subport} is a text styling library and part of gettext
+    long_description        ${subport} provides an easy way to add styling to programs that \
+                            produce output to a console or terminal emulator window. \
+                            The library is part of gettext and a prerequisite for tools like \
+                            msgfmt and friends.
+
+    depends_lib             port:libiconv \
+                            port:ncurses
+    worksrcdir              ${distname}/libtextstyle
+}
+
+if {$subport eq $name} {
+    long_description        ${name} provides the tools like msgfmt for writing multilanguage applications. \
+                            The tools are required during the build of applications which use \
+                            the gettext system for multilanguage support. The libraries for \
+                            runtime (libintl) are provided via the gettext-runtime subport.
+
+    depends_lib-append      port:gettext-libtextstyle \
+                            port:gettext-runtime
+
+    worksrcdir              ${distname}/gettext-tools
+
+    # disable libasprintf to make test lang-c++ check not fail
+    configure.args-append   --with-installed-libtextstyle \
+                            --disable-libasprintf
+
+    # The gettext-tools installation also installs libintl again. Here
+    # I remove the libintl parts because they are provided by the runtime
+    # subport gettext-runtime
```

Comment:
Surely there must be a way to tell the build system not to build these again.

Reply:
The worksrcdir is the gettext-tools subdirectory. So this should not build everything again. I had a look at the fink friends and it seems that the tools required some parts from runtime also. But this is how gettext is supposed to be build.